component restructuring and Interactor implemented (panning & Trackball)

laurabuenemann opened the pull request.

M4anuel committed:
component restructuring and Interactor implemented (panning & Trackball)

michael-duer commented on backend/src/dicoms/dicom_slice.js:
```diff
@@ -45,7 +45,7 @@ export const DicomSlice = async (req, res, models) => {
     console.log({ FilePath, position });
     var spawn = require("child_process").spawn;
-    var process = spawn("python3", [
+    var process = spawn("python", [
```
Hier würde ich die Logik von Manuel wieder einfügen, damit automatisch geschaut wird ob man python oder python3 benötigt.

M4anuel replied:
Die hatte einige Bugs und habe sie deswegen von den meisten Branches entfernt. Das sollte aber standardmässig python3 sein

michael-duer reviewed (commented):
Sieht gut aus! Was noch fehlt, ist das Anpassen der vorhandenen Tests an die neuen Components sowie deren Umbenennung.

M4anuel committed:
renaming and setup of fetch test

M4anuel commented:
Habe einen Test angefangen aufzusetzen in welchem ich das backend mocke, habe aber noch Probleme mit der JsonDcm Klasse (glaube ich). Habe auf jeden Fall den Komponenten renamed. Den alten habe ich noch nicht herausgenommen, vielleicht kannst du das machen und anschauen ob du herausfindes was beim Test fehlt? @michael-duer

michael-duer committed:
Refactor tests and clean up unused files

- Fix issues in fetchData.test.js ensuring all tests pass correctly
- Add tests for DicomViewer in DicomViewer.test.js
- Remove obsolete files from the project (ImageViewerContainer)
- Minor adjustments

michael-duer commented:
Die Tests für fetchData und DicomViewer funktionieren nun alle. Zudem habe ich nicht mehr benötigte Dateien wie den ImageViewerContainer entfernt.

michael-duer committed:
Refactor file structure

- Move ProgressBar.js and DicomImageRenderer.js from DicomImageRenderer to DicomViewer folder to centralize related components
- Remove DicomImageRenderer folder
